xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -613,6 +613,39 @@ public URL createAttachmentURL(String filename, String spaces, String name, Stri
             }
         }
 
+        if (isContextDoc(xwikidb, spaces, name, context)) {
+            XWikiAttachment attachment = context.getDoc().getAttachment(filename);
+            if (attachment != null) {
+                return createAttachmentRevisionURL(filename, spaces, name, attachment.getVersion(),
+                    querystring, xwikidb, context);
+            } else {
+                if (LOGGER.isErrorEnabled()) {
```

Comment:
You shouldn't use this with SLF4J when there's only a single statement.

Reply:
Good to know I actually replicate this from https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java#L610 so I guess I can refactor this one too.

Reply:
It's the same cost as the "IF" with SLF4J (that's why the api has positional parameters btw ;)). So it just makes the code harder to read than needed.

Ofc if you have to do some costly computations to perform the logging then yes it can be useful.